pkg/fqdn/cache.go

```diff
@@ -770,6 +782,18 @@ func NewDNSZombieMappings(max, perHostLimit int) *DNSZombieMappings {
 	}
 }
 
+// Names returns the FQDN entries referred to in the DNSZombieMappings
+func (zombies *DNSZombieMappings) Names() []string {
+	zombies.Lock()
+	defer zombies.Unlock()
+
+	names := make([]string, 0, len(zombies.deletes))
+	for _, zombie := range zombies.deletes {
+		names = append(names, zombie.Names...)
+	}
+	return slices.Unique(names)
```

Comment:
Since the `Names` is really just needed for this use-case, perhaps better to just pass in the map to which to collect the names into. Will think about it.